x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go

```diff
@@ -253,3 +253,7 @@ func (f *fleetGateway) stop() {
 	close(f.done)
 	f.wg.Wait()
 }
+
+func (f *fleetGateway) SetClient(client clienter) {
+	f.client = client
```

Comment:
maybe we should here and in acker make sure client is not change while performing execute/ack action.
at least it can be misleading e.g 
- performin a call using client1 - fails
- client is updated
- reporting failure from step 1 but client.URI is different now

Reply:
Actually I think this flow is correct, because each action from Fleet is handled synchronously in the Agent. So when a policy change comes in with an updated hosts, they action will not be Ack'd until this code is able to re-connect back to Kibana using the new hosts information.

`fleetapi.NewAuthWithConfig` ensures that the created client can communicate with Kibana, so that means the Ack will not happen until the updated client is created and set.